pkg/publish/options.go

```diff
@@ -78,6 +78,15 @@ func WithAuthFromKeychain(keys authn.Keychain) Option {
 	}
 }
 
+// WithKeyChain is a functional option for overriding the default
+// authenticator on a default publisher using an authn.Keychain
+func WithKeyChain(keys authn.Keychain) Option {
```

Comment:
We don't need to add a new method, we can just keep using `WithAuthFromKeychain`.